central/imageintegration/datastore/datastore_impl.go

```diff
@@ -89,3 +102,21 @@ func (ds *datastoreImpl) RemoveImageIntegration(ctx context.Context, id string)
 
 	return ds.storage.Delete(ctx, id)
```

Comment:
You probably want to remove the item from the index and the storage here, and push the item to the index on Addition, Update or Upsert.
The pattern I saw in `alert` is
- For Add/Update/Upsert:
```
if err := ds.storage.CUfunction(ctx, object); err != nil { return err }
if err := ds.index.WriteToIndex(objectToIndex); err != nil { return err }
return ds.storage.AckKeysIndexed(ctx, object.GetId())
```
- For deletion:
```
errorList := errorhelpers.NewErrorList("deleting")
if err := ds.storage.Delete(ctx, objId); err != nil { errorList.AddError(err) }
if err := ds.indexer.Delete(objid); err != nil { errorList.AddError(err) }
if err := ds.storage.AckKeysIndexed(ctx, objid); err != nil { errorList.AddError(err) }
return errorList.ToError()
```

Reply:
I could be wrong. But AckKeysIndexed is not available for Bolt as far as I've seen.